src/Microsoft.AspNetCore.Razor.Tasks/RazorGenerate.cs

```diff
@@ -3,47 +3,74 @@
 
 using System.Text;
 using Microsoft.Build.Framework;
-using Microsoft.CodeAnalysis.CommandLine;
 
 namespace Microsoft.AspNetCore.Razor.Tasks
 {
     public class RazorGenerate : DotNetToolTask
     {
         [Required]
-        public string[] Sources { get; set; }
+        public ITaskItem[] Inputs { get; set; }
 
         [Required]
         public string ProjectRoot { get; set; }
 
-        [Required]
-        public string OutputPath { get; set; }
-
         [Required]
         public string TagHelperManifest { get; set; }
 
         internal override string Command => "generate";
 
+        protected override bool ValidateParameters()
+        {
+            for (var i = 0; i < Inputs.Length; i++)
+            {
+                if (!EnsureRequiredMetadata(Inputs[i], "GeneratedOutput") ||
+                    !EnsureRequiredMetadata(Inputs[i], "ViewEnginePath"))
+                {
+                    return false;    
+                }
+            }
+
+            return base.ValidateParameters();
+        }
+
         protected override string GenerateResponseFileCommands()
         {
             var builder = new StringBuilder();
 
             builder.AppendLine(Command);
 
-            for (var i = 0; i < Sources.Length; i++)
+            for (var i = 0; i < Inputs.Length; i++)
             {
-                builder.AppendLine(Sources[i]);
+                var input = Inputs[i];
+                builder.AppendLine("-s");
```

Comment:
In case of CommandOptionType.MultipleValue, do you need to specify this option with every value?